ts/Stock/StockToolsBindings.ts

```diff
@@ -595,112 +625,63 @@ extend(NavigationBindings.prototype, {
-     * space.
+     * If axis is removed, the current plot area streaches to fit into 100%
+     * of the plot area.
      *
      * @private
      * @function Highcharts.NavigationBindings#resizeYAxes
-     * @param {number} [defaultHeight]
-     * Default height for yAxis
+     * @param {string} [removedYAxisHeight]
+     *
+     *
      */
     resizeYAxes: function (
         this: Highcharts.StockToolsNavigationBindings,
-        defaultHeight?: number
+        removedYAxisHeight?: string
     ): void {
-        defaultHeight = defaultHeight || 20; // in %, but as a number
+        // The height of the new axis before rescalling. In %, but as a number.
+        const defaultHeight = 20;
         var chart = this.chart,
             // Only non-navigator axes
             yAxes = chart.yAxis.filter(bindingsUtils.isNotNavigatorYAxis),
             plotHeight = chart.plotHeight,
-            allAxesLength = yAxes.length,
             // Gather current heights (in %)
-            positions = this.getYAxisPositions(
+            { positions, allAxesHeight } = this.getYAxisPositions(
                 yAxes,
                 plotHeight,
-                defaultHeight
+                defaultHeight,
+                removedYAxisHeight
             ),
-            resizers = this.getYAxisResizers(yAxes),
-            allAxesHeight: number = (positions as any).allAxesHeight,
-            changedSpace = defaultHeight;
-
-        // More than 100%
-        if (allAxesHeight > 1) {
-            // Simple case, add new panes up to 5
-            if (allAxesLength < 6) {
-                // Added axis, decrease first pane's height:
-                positions[0].height = correctFloat(
-                    positions[0].height - changedSpace
-                );
-                // And update all other "top" positions:
-                positions = this.recalculateYAxisPositions(
-                    positions,
-                    changedSpace
-                );
-            } else {
-                // We have more panes, rescale all others to gain some space,
-                // This is new height for upcoming yAxis:
-                defaultHeight = 100 / allAxesLength;
-                // This is how much we need to take from each other yAxis:
-                changedSpace = defaultHeight / (allAxesLength - 1);
-
-                // Now update all positions:
-                positions = this.recalculateYAxisPositions(
-                    positions,
-                    changedSpace,
-                    true,
-                    -1
-                );
-            }
-            // Set last position manually:
-            positions[allAxesLength - 1] = {
-                top: correctFloat(100 - defaultHeight),
-                height: defaultHeight
-            };
+            resizers = this.getYAxisResizers(yAxes);
 
+        if (
+            !removedYAxisHeight &&
+            allAxesHeight <= correctFloat(0.8 + defaultHeight / 100)
+        ) {
```

Comment:
Could you please add a comment here? I'm not sure what is the purpose of this if-else